### ISSUE TITLE
model: codingjepa.modules.encoder (6-layer Transformer + RoPE + special tokens)

### ISSUE THREAD

**Issue by AbdelStark** [AbdelStark/CodingJEPA#58]
## Context

RFC-0003 §D1: 6-layer pre-norm Transformer, hidden 512, 8 heads, FFN 2048, GELU, dropout 0.1, RoPE positional encoding, max sequence length 512 tokens, ~30M params, output is `[CLS]` hidden state.

## Scope

- `codingjepa.modules.encoder.Encoder` — `nn.Module` per RFC-0003 §D1.
- Special tokens: `[CLS]`, `[SEP]`, `[PAD]`, `[CHUNK]`, `[I_*]`, `[I_NONE]`.
- Forward: `(B, T) -> (B, D)` returning the `[CLS]` hidden state (mirrors LeWM `output.last_hidden_state[:, 0]`).
- RoPE applied per layer; no learned absolute positional embedding.

## Out of Scope

Projector / predictor (separate issues).

## Design Reference

- RFC: [RFC-0003 §D1](../blob/main/docs/rfcs/RFC-0003-encoder-predictor-decoder-stack.md)

## Acceptance Criteria

- [ ] `tests/modules/test_encoder_shape.py` asserts forward shapes and parameter count ≈ 30M ± 5%.
- [ ] `tests/modules/test_rope.py` verifies the rotary encoding is applied (basic sin/cos pattern check).
- [ ] `tests/modules/test_special_tokens.py` asserts `[CLS]` token id is 2.

## Dependencies

- Depends on: #16

## Metadata

```
type: feature
area: model
priority: p0
effort: m
spec: rfc-0003
milestone: v0.2
tracking: #5
```
